lib/libspl/include/umem.h

```diff
@@ -137,6 +137,12 @@ umem_free(const void *ptr, size_t size __maybe_unused)
 	free((void *)ptr);
 }
 
+static inline void
+umem_free_aligned(void *ptr, size_t size __maybe_unused)
```

Comment:
There is no need to invent posix_memalign_free(), which looks like a standard library function. It is confusing since the documentation says that allocations from posix_memalign() should be passed to the free function.

I would prefer to have a macro redirect this to umem_free and have a comment saying that umem_free_aligned was made for supporting portability with non-POSIX platforms that require a different free to be used with aligned allocations.

Reply:
I've gotten rid of this and replaced it with a define on POSIX systems